x/fswap/spec/01_concepts.md

```diff
@@ -29,10 +43,11 @@ type Config struct {
 }
 ```
 
-## Proposal
+## MsgSetSwap
 
-Typically, a `Swap` is proposed and submitted through foundation via a `MsgSetSwap`.
-This proposal prescribes to the standard foundation process. If the proposal passes, the `Swap` can be used on chain.
+Typically, a `Swap` is proposed and submitted through `x/foundation` via a `MsgSetSwap`.
+Other modules (`x/gov`, `x/foundation`) will include this message in their proposals. If the proposal passes, the `Swap` can be used on chain.
+`ToDenomMetadata` is [`Metadata`](../../bank/types/bank.pb.go#L325) in `x/bank` module, and it MUST meet these [limitations](../../bank/types/metadata.go#L11).
```

Comment:
+1 for the reference to the x/bank denom metadata restrictions. You should also state the restrictions by x/swap.

Reply:
I mean, we should state the restrictions by x/swap, i.e. one cannot override existing denom metadata, etc.

Reply:
> one cannot override existing denom metadata

This can be modified by config, doesn't it?

Reply:
No, it's not allowed. You should also state about the configuration.